core/string/modulo_spec.rb

```diff
@@ -14,9 +14,9 @@
     ("%d%% %s" % [10, "of chickens!"]).should == "10% of chickens!"
   end
 
-  it "formats single % character at the end as literal %" do
-    ("%" % []).should == "%"
-    ("foo%" % []).should == "foo%"
+  it "raises an error if single % appears at the end" do
+    lambda { ("%" % []) }.should raise_error(ArgumentError)
+    lambda { ("foo%" % [])}.should raise_error(ArgumentError)
   end
```

Comment:
If the change is merged, it will only applies to `trunk` initially, and then maybe be backported.
Therefore, could you add a version guard like this
```ruby
ruby_version_is ""..."2.5" do
  old spec
end
ruby_version_is "2.5" do
  new spec
end
```
?